packages/vite-plugin/src/client/es/hmr-client-worker.ts

```diff
@@ -34,6 +34,15 @@ self.addEventListener('fetch', (fetchEvent) => {
  * https://bugs.chromium.org/p/chromium/issues/detail?id=1247690#c_ts1631117342
  */
 async function sendToServer(url: URL): Promise<Response> {
+	if (url.href.includes('/_favicon/')) {
+		const response = await fetch(url.href)
+		return new Response(response.body, {
+			headers: {
+				'Content-Type': response.headers.get('Content-Type') ?? 'text/javascript',
+			},
+		})
+	}
```

Comment:
I imagine there are other use cases to not redirect to the dev server. Let's move `url.href.includes('/_favicon/')` up into line 23 and abstract it into a guard function.

Reply:
Agree, I put them in a array `const ignoreURLPaths = ['/_favicon/']`, in case there are other paths to ignore as well (also perhaps allow users to add their own ignore paths). It's not as efficient as doing separate if statements as before (though V8 is probably optimizing these on its own), but as I mentioned in the future users might want to add other ignore paths?